### PR TITLE
feat(tier4_autoware_utils): add new orientation calculation

### PR DESCRIPTION
**Pull request by purewater0901**
Signed-off-by: yutaka <purewater0901@gmail.com>

## Description
Add a new option to calculate the interpolated pose orientation

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/tier4_autoware_utils/test/src/geometry/test_geometry.cpp [hunk 1]
```diff
@@ -874,45 +874,102 @@ TEST(geometry, calcInterpolatedPose)
   }
 
   // Quaternion Interpolation
-  {
-    geometry_msgs::msg::Pose src_pose;
-    src_pose.position = createPoint(0.0, 0.0, 0.0);
-    src_pose.orientation = createQuaternionFromRPY(deg2rad(0), deg2rad(0), deg2rad(0));
+  {{geometry_msgs::msg::Pose src_pose;
```
Comment (TakaHoribe) [inline common/tier4_autoware_utils/test/src/geometry/test_geometry.cpp:877]: Unusual shape, but CI has passed...hmm

Comment (kenji-miyake) [reply common/tier4_autoware_utils/test/src/geometry/test_geometry.cpp:877]: He asked me about that in another PR. Removing one pair of `{}` can solve this format collapse.

Comment (purewater0901) [reply common/tier4_autoware_utils/test/src/geometry/test_geometry.cpp]: Fixed

**Review by TakaHoribe — APPROVED** [review]
LGTM. Note, this function is still a new one, not used anywhere.

**Pull request merged**